Sources/Event Logging/TracksService.m

```diff
@@ -161,7 +161,7 @@ - (void)sendQueuedEvents
     [self.timer invalidate];
     [[NSNotificationCenter defaultCenter] postNotificationName:TrackServiceWillSendQueuedEventsNotification object:nil];
     
-    NSArray *events = [self.tracksEventService allTracksEvents];
+    NSArray *events = [self.tracksEventService tracksEventsWithLimit:1000];
```

Comment:
My first reaction seeing this change was "if we are batching, what happens to any other batch at this time? Is this safe or will it lose events? do we need to manage the next batch(es)?"

Correct me if I'm wrong given you're surely fresher on the internal details than I am.

I think this is 100% safe without any logic to get the next batch(es). That's because any remaining even will be picked up on the next `sendQueuedEvents` call, and that will continue until the even queue will be empty.

Reply:
Yeah, they should just wait around until the next time we send events, and then we'll send the next batch.

There's _potentially_ an issue if there's a problem with one particular Tracks event that causes the batch to be rejected; that batch would remain as the first batch, and keep retrying indefinitely, preventing later events from being sent. But I don't know if that's even a possibility, and if it is, it's no worse than the situation we're _already_ in — a single invalid event would _already_ cause all the other events to fail, even without that change. So I'm not concerned about fixing that in this PR.